express-mutations/server.ts

```diff
@@ -0,0 +1,111 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/pagila',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+app.use(express.json());
+
+// Endpoint for testing
+app.get('/api/actors/:actorId', async (req, res, next) => {
+  try {
+    const { actorId } = req.params;
+    if (!Number.isInteger(+actorId)) {
+      throw new ClientError(400, `Non-integer actorId: ${actorId}`);
+    }
+    const sql = `
+      select * from "actors"
+      where "actorId" = $1;
+    `;
+    const params = [actorId];
+    const result = await db.query(sql, params);
+    const actor = result.rows[0];
+    if (!actor) throw new ClientError(404, `actor ${actorId} not found`);
+    res.json(actor);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/actors', async (req, res, next) => {
+  try {
+    const newActor = req.body;
+    const sql = `
+      insert into
+        "actors"
+        ("firstName",
+        "lastName")
+        values ($1, $2)
+        returning *;
+      `;
+    const params = [newActor.firstName, newActor.lastName];
+    const result = await db.query(sql, params);
+    res.status(201).json(result.rows[0]);
+    if (!newActor) {
```

Comment:
You should do this check _before_ you issue the query. Otherwise you will be trying to insert an invalid row into the database. Also, you are only checking if there is a body. You are not checking that the firstName and lastName fields are present.